{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentTemplate.json#",
  "contentVersion": "1.0.0.0",
  "Parameters": {
    "Workspace": {
      "type": "string"
    }
  },
  "resources": [
    {
      "id": "[concat(resourceId('Microsoft.OperationalInsights/workspaces/providers', parameters('workspace'), 'Microsoft.SecurityInsights'),'/alertRules/4595192f-671b-4724-aa62-093a9724c2f3')]",
      "name": "[concat(parameters('workspace'),'/Microsoft.SecurityInsights/4595192f-671b-4724-aa62-093a9724c2f3')]",
      "type": "Microsoft.OperationalInsights/workspaces/providers/alertRules",
      "kind": "Scheduled",
      "apiVersion": "2021-03-01-preview",
      "properties": {
        "queryPeriod": "5m",
        "queryFrequency": "5m",
        "query": "Veeam_GetSecurityEvents\n| where instanceId == 41800\n| extend Endpoint = extract(\"param2=\\\"([^\\\"]*)\\\"\", 1, SyslogMessage)\n| project\n    Date = format_datetime(TimeGenerated, 'dd.MM.yyyy HH:mm'),\n    DataSource = original_host,\n    EventId = instanceId,\n    [\"User Name\"] = user,\n    [\"Endpoint\"] = Endpoint,\n    MessageDetails = Description,\n   Severity = SeverityDescription",
        "description": "Detects failed backup operations. This might indicate system or storage issues, or a potential sabotage of the backup infrastructure.",
        "tactics": [
          "DefenseEvasion",
          "Impact"
        ],
        "status": "Available",
        "severity": "High",
        "triggerThreshold": 0,
        "triggerOperator": "gt",
        "name": "Attempt to Delete Backup Failed",
        "customDetails": {
          "Severity": "Severity",
          "Date": "Date",
          "EventId": "EventId",
          "MessageDetails": "MessageDetails",
          "VbrHostName": "DataSource"
        },
        "eventGroupingSettings": {
          "aggregationKind": "AlertPerResult"
        }
      }
    }
  ]
}
